packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx

```diff
@@ -109,7 +109,26 @@ export class DropdownBase extends BaseComponent<IDropdownInternalProps, IDropdow
   public componentWillReceiveProps(newProps: IDropdownProps): void {
     // In controlled component usage where selectedKey is provided, update the selectedIndex
     // state if the key or options change.
-    const selectedKeyProp: keyof IDropdownProps = this.props.multiSelect ? 'selectedKeys' : 'selectedKey';
+    let selectedKeyProp: 'defaultSelectedKeys' | 'selectedKeys' | 'defaultSelectedKey' | 'selectedKey';
+
+    // this does a shallow compare (assumes options are pure), for the purposes of determining whether
+    // defaultSelectedKey/defaultSelectedKeys are respected.
+    const didOptionsChange = newProps.options !== this.props.options;
+
+    if (this.props.multiSelect) {
```

Comment:
Why `this.props.multiSelect` not `newProps.multiSelect`? (note that I haven't looked at Dropdown's implementation in detail yet, so I don't actually know which one is "right" here--just something to consider)

Reply:
Super valid question. Thanks for pointing out. I was following what the current logic did.

Reply:
It's very likely the existing logic is buggy..

Reply:
If the existing logic's use of this.props was intentional, my best guess as to the reason is that maybe they wanted to prohibit changing between single and multi select (except that wouldn't have worked as intended since newProps is later saved to this.props).